pkg/chunk/prefetch.go

```diff
@@ -39,23 +39,23 @@ func newPrefetcher(parallel int, fetch func(string)) *prefetcher {
 
 func (p *prefetcher) do() {
 	for key := range p.pending {
-		p.Lock()
-		if _, ok := p.busy[key]; !ok {
-			p.busy[key] = true
-			p.Unlock()
-
-			p.op(key)
+		p.op(key)
 
-			p.Lock()
-			delete(p.busy, key)
-		}
+		p.Lock()
+		delete(p.busy, key)
 		p.Unlock()
 	}
 }
 
 func (p *prefetcher) fetch(key string) {
+	p.Lock()
```

Comment:
this may be blocked and fetch should not block the read process

Reply:
Which line it will be blocked on?